WordPress/gutenberg
# Remove "Enable Comments" button from Post Comments Form block

## What?
Fixes https://github.com/WordPress/gutenberg/issues/54019

This PR removes the "Enable Comments" button from the Post Comments Form block in scenarios where comments are disabled or unsupported for the current post type or post.

## Why?

The button created confusion in the site editor when it appeared in contexts where enabling comments was not actionable. Removing it simplifies the UI and ensures users are only presented with relevant information.

## How?

The "Enable Comments" button has been removed, and warnings are displayed to inform users when comments are disabled or unsupported. The placeholder block consistently shows the status of comments without unnecessary UI elements.

## Testing Instructions

1. Enable TT3 theme.
2. Open the site editor and add a template containing the Post Comments Form block.
3. Test with comments enabled and disabled globally, and for specific post types.
4. Confirm that the button is no longer present, and warnings appear as expected when comments are disabled.

## Screenshots or screencast 

## Before

![comments_button_enabled](https://github.com/user-attachments/assets/9ab4c891-9526-43fd-836e-5e5267936424)


## After


![enable_comments_disabled](https://github.com/user-attachments/assets/a2c362e7-99df-477d-94b4-1b8199fdd181)
